Assets/Aptabase/Scripts/Aptabase.cs

```diff
@@ -0,0 +1,140 @@
+            { "EU", "https://eu.aptabase.com" },
+            { "DEV", "http://localhost:3000" },
+            { "SH", "" },
+        };
+        
+        private const string EVENT_ENDPOINT = "/api/v0/event";
+        private const string SDK_VERSION = "0.0.1";
+
+        private static AptabaseSettings _settings;
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+        private static void Initialize()
+        {
+            //load settings
+            _settings = Resources.Load<AptabaseSettings>("AptabaseSettings");
+            if (_settings == null)
+            {
+                Debug.LogWarning("Aptabase Settings not found. Tracking will be disabled");
+                return;
+            }
+            
+            var key = _settings.AppKey;
+            
+            var parts = key.Split("-");
+            if (parts.Length != 3 || !_hosts.ContainsKey(parts[1]))
+            {
+                Debug.LogWarning($"The Aptabase App Key {key} is invalid. Tracking will be disabled");
+                return;
+            }
+            
+            _baseURL = GetBaseUrl(parts[1]);
+        }
+        
+        private static string GetBaseUrl(string region)
+        {
+            if (region == "SH")
+            {
+                if (string.IsNullOrEmpty(_settings.SelfHostURL))
+                {
+                    Debug.LogWarning("Host parameter must be defined when using Self-Hosted App Key. Tracking will be disabled.");
+                    return null;
+                }
+
+                return _settings.SelfHostURL;
+            }
+
+            return _hosts[region];
+        }
+
+        public static void TrackEvent(string eventName, Dictionary<string, object> props = null)
+        {
+            SendEvent(eventName, props);
+        }
+
+        private static async void SendEvent(string eventName, Dictionary<string, object> props)
+        {
+            if (string.IsNullOrEmpty(_baseURL))
+                return;
+            
+            try
+            {
+                var now = DateTime.UtcNow;
+                var timeSince = now.Subtract(_lastTouched);
+                if (timeSince >= _sessionTimeout)
+                    _sessionId = NewSessionId();
+
+                _lastTouched = now;
+
+                props ??= new Dictionary<string, object>();
+                
+                //create the main dictionary for EventData
+                var eventData = Json.Serialize(new Dictionary<string, object>(5)
+                {
+                    { "timestamp", DateTime.UtcNow.ToString("o") },
+                    { "sessionId", _sessionId },
+                    { "eventName", eventName },
+                    { "systemProps", new Dictionary<string, object>(7)
+                    {
+                        { "isDebug", Application.isEditor || Debug.isDebugBuild },
+                        { "osName", SystemInfo.operatingSystemFamily.ToString() },
```

Comment:
It seems like this property is a bit limited in terms what value it returns, doesn't support consoles for example. Any idea if there's anything better that would return other values like Switch, PS4, etc..?

Reply:
Yeah, definitely looks much a much better option!